dubbo-metadata-report/dubbo-metadata-definition/src/main/java/org/apache/dubbo/metadata/definition/TypeDefinitionBuilder.java

```diff
@@ -35,20 +36,16 @@
  */
 public class TypeDefinitionBuilder {
 
-    private static final ThreadLocal<ArrayList<TypeBuilder>> builders;
+    private static final TypeBuilder ARRAY_BUILDER = new ArrayTypeBuilder();
+    private static final TypeBuilder COLLECTION_BUILDER = new CollectionTypeBuilder();
+    private static final TypeBuilder MAP_BUILDER = new MapTypeBuilder();
+    private static final TypeBuilder ENUM_BUILDER = new EnumTypeBuilder();
 
-    static {
-        builders = new ThreadLocal<ArrayList<TypeBuilder>>();
-        builders.set(new ArrayList<TypeBuilder>());
-        builders.get().add(new ArrayTypeBuilder());
-        builders.get().add(new CollectionTypeBuilder());
-        builders.get().add(new MapTypeBuilder());
-        builders.get().add(new EnumTypeBuilder());
-    }
+    private static final List<TypeBuilder> BUILDERS = Arrays.asList(ARRAY_BUILDER, COLLECTION_BUILDER, MAP_BUILDER, ENUM_BUILDER);
```

Comment:
Can you explain why you are removing `ThreadLocal`?

Reply:
Hi, huxing. Sorry, I missed your message before.

I checked the implementation of the associated TypeBuilder and it seems that they are all thread-safe (no global variables). So we just need to use static final decoration.